### ISSUE TITLE
False positive: Unexpected token ,

### ISSUE THREAD

**Issue by josephzidell** [eslint/eslint#11643]
**Tell us about your environment**

* **ESLint Version: v4.19.1**
* **Node Version: v8.11.1**
* **npm Version: 6.8.0**

**What parser (default, Babel-ESLint, etc.) are you using? ESLint**

**Please show your full configuration:**
Default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
addNote: function (type, note) {
	this.notes.push({type, note});
}
```

**What did you expect to happen?**
Everything should be fine

**What actually happened? Please include the actual, raw output from ESLint.**
```
Unexpected token ,
```

**Are you willing to submit a pull request to fix this bug?**
Possibly

**Comment by platinumazure** [comment]
Hi @josephzidell, thanks for the issue.

Could you please provide a complete reproduction? With the code you have shown, I get `Unexpected token function`, because it is parsing the `addNote:` as the beginning of a labeled statement.

Also, when I turn this into a simple object assignment, I get no parser error:

```js
const obj = {
    addNote: function (type, note) {
	this.notes.push({type, note});
    }
};
```

Please include everything about your configuration (especially what parser you are using). Thanks!

**Comment by josephzidell** [comment]
Hi @platinumazure 

I'm using ESLint v4.19.1. My full config is
```json
{
  "root": true,
  "extends": "eslint:recommended",
  "globals": {
    "__WEEX__": true
  },
  "rules": {
    "no-console": "off",
    "no-useless-escape": 0,
    "quotes": ["error", "single"],
    "no-undef": 0,
    "no-unused-vars": 0,
    "no-trailing-spaces": "error",
    "space-in-parens": ["error", "never"],
    "spaced-comment": ["error", "always", { "markers": ["end"] }],
    "space-before-blocks": "error",
    "semi-spacing": ["error", {"before": false, "after": true}],
    "keyword-spacing": "error",
    "array-bracket-spacing": ["error", "never"],
    "comma-spacing": ["error", { "before": false, "after": true }]
  }
}
```

It's running within [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) using the [ESLintSniff](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Generic/Sniffs/Debug/ESLintSniff.php)

When using `const obj`, it gives
```
1 | ERROR | eslint says: Parsing error: The keyword 'const' is reserved
```

If I change to `var obj`, then it gives
```
3 | ERROR | eslint says: Parsing error: Unexpected token ,
```

I hope this helps 🤞

**Comment by g-plane** [comment]
Try adding lines below to your configuration:

```json
  "ecmaVersion": 2015,
```

**Comment by josephzidell** [comment]
@g-plane Like a charm :)
